### PR TITLE
Add ability to configure storage to scorecard

### PR DESCRIPTION
**Pull request by fabianvf**
**Description of the change:**
depends on https://github.com/operator-framework/api/pull/135
**Motivation for the change:**

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: internal/scorecard/storage.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		defer outStream.Close()
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  nil,
+			Stdout: outStream,
+			Stderr: &stderr,
```
Comment (estroz) [inline internal/scorecard/storage.go:73]: stderr gets thrown away here I think. This should be returned and set in test output if not empty.

File: internal/scorecard/storage.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+		evaledPath, err := filepath.EvalSymlinks(baseName)
+		if err != nil {
+			return err
+		}
+
+		if mode&os.ModeSymlink != 0 {
+			linkname := header.Linkname
+
+			if !filepath.IsAbs(linkname) {
+				_ = filepath.Join(evaledPath, linkname)
```
Comment (estroz) [inline internal/scorecard/storage.go:123]: What's this for?

File: testdata/ansible/memcached-operator/bundle/tests/scorecard/config.yaml [hunk 3]
```diff
@@ -12,38 +12,59 @@ stages:
     labels:
       suite: basic
       test: basic-check-spec-test
+    storage:
```
Comment (estroz) [inline testdata/ansible/memcached-operator/bundle/tests/scorecard/config.yaml:15]: The storage spec has `omitempty` on it, so these shouldn't be showing up here.

File: internal/scorecard/scorecard.go [hunk 4]
```diff
@@ -195,13 +200,20 @@ func (r PodTestRunner) Cleanup(ctx context.Context) (err error) {
...
 	return nil
 }
 
 // RunTest executes a single test
 func (r PodTestRunner) RunTest(ctx context.Context, test v1alpha3.TestConfiguration) (*v1alpha3.TestStatus, error) {
+
 	// Create a Pod to run the test
 	podDef := getPodDefinition(r.configMapName, test, r)
+
+	if test.Storage.Spec.MountPath.Path != "" {
```
Comment (estroz) [inline internal/scorecard/scorecard.go:213]: I don't see the top-level [`v1alpha3.Configuration.Storage`](https://github.com/operator-framework/api/blob/master/pkg/apis/scorecard/v1alpha3/configuration_types.go#L24) field being respected here.

File: internal/scorecard/storage.go (hunk 2 above)
Comment (fabianvf) [reply internal/scorecard/storage.go:123]: looks like nothing to me, I can't imagine this is doing anything

File: internal/scorecard/scorecard.go (hunk 4 above)
Comment (fabianvf) [reply internal/scorecard/scorecard.go:213]: Good catch, it looks like this is the first field on the scorecard config that needs to be propagated to individual tests

File: testdata/ansible/memcached-operator/bundle/tests/scorecard/config.yaml (hunk 3 above)
Comment (fabianvf) [reply testdata/ansible/memcached-operator/bundle/tests/scorecard/config.yaml:15]: odd... It looks like omitempty is missing from the `spec` field, could that be causing this? sanity check won't pass with these removed, might need to followup to add that in api, would this be blocking?

File: internal/scorecard/storage.go (hunk 1 above)
Comment (fabianvf) [reply internal/scorecard/storage.go]: great catch

**Review by theishshah — APPROVED** [review]

File: testdata/ansible/memcached-operator/bundle/tests/scorecard/config.yaml (hunk 3 above)
Comment (estroz) [reply testdata/ansible/memcached-operator/bundle/tests/scorecard/config.yaml:15]: Yeah that looks correct. Not a blocker, just needs a follow-up.

**Review by estroz — APPROVED** [review]
/lgtm

**Pull request merged**